File: _typeorm/src/entity/Review.ts
import {Entity, PrimaryColumn, Column, ManyToOne, JoinColumn,
        Index} from "typeorm";
import {User} from './User'
import {Movie, MovieView} from './Movie'


@Entity()
export class Review {

    // PrimaryGeneratedColumn ignores id even if specified
    @PrimaryColumn()
    id: number;

    @Column()
    body: string;

    @Column()
    rating: number;

    @Column()
    creation_time: Date;

    @Column({select: false})
    @Index()
    movie_id: number;

    @Column({select: false})
    @Index()
    author_id: number;

    @ManyToOne(type => User, user => user.reviews)
    @JoinColumn({name: 'author_id'})
    author: User

    @ManyToOne(type => Movie, movie => movie.reviews)
    @JoinColumn({name: 'movie_id'})
    movie: Movie
}

File: _dotnet/EdgeDB.Net.IMDBench/Benchmarks/EFCore/Models/Review.cs
using System;
using System.Collections.Generic;

namespace EdgeDB.Net.IMDBench.Benchmarks.EFCore.Models;

public partial class Review
{
    public int Id { get; set; }

    public string Body { get; set; } = null!;

    public int Rating { get; set; }

    public DateTime CreationTime { get; set; }

    public int AuthorId { get; set; }

    public int MovieId { get; set; }

    public virtual User Author { get; set; } = null!;

    public virtual Movie Movie { get; set; } = null!;
}
